### PR TITLE
[doc] Release notes update for 0.12

### PR DESCRIPTION
**Pull request by sdaftuar**

### CODE REVIEW

**Comment by sdaftuar** [comment]
@jonasschnelli Let me know if I got the banlist stuff right...

File: doc/release-notes.md [hunk 1]
```diff
@@ -134,12 +137,32 @@ Replace-by-fee transactions
...
+
+The `-permitrbf` option is *not recommended* for wallet users seeking to avoid
+receipt of unconfirmed opt-in transactions, because this option does not
+prevent transactions which are replaceable under BIP 125 from being accepted
+(only subsequent replacements, which other nodes on the network that implement
+BIP 125 are likely to relay and mine).  Wallet users wishing to detect whether
+a transaction is subject to replacement under BIP 125 should instead use the
+updated RPC calls `gettransaction` and `listtransactions`, which now have an
+additional field in the output indicating if a transaction is replaceable under
+BIP125 ("bip125-replaceable")".
```
Comment (MarcoFalke) [inline doc/release-notes.md]: Nit: trailing "?

**Comment by MarcoFalke** [comment]
ACK  8691c39

File: doc/release-notes.md (hunk 1 above)
Comment (sdaftuar) [inline doc/release-notes.md]: Oops, thanks for catching, will fix.

**Comment by sdaftuar** [comment]
Fixed @MarcoFalke's nit and squashed

**Comment by jonasschnelli** [comment]
ACK!
Thanks @sdaftuar for the banlist part! It short and perfect.

**Comment by MarcoFalke** [comment]
re-ACK 65d384f. I appreciate that this is addressing my concern from #7391. (Expose the option to non-miners)

**Comment by achow101** [comment]
ACK

**Comment by petertodd** [comment]
utACK https://github.com/sdaftuar/bitcoin/commit/65d384fe82fcfbe060f1c4884572cf9439c9f840

I'll do a separate pull-req for it, but I think the release notes need to make clear that opt-in or not, unconfirmed transactions are easy to replace. (see https://petertodd.org/2016/are-wallets-ready-for-rbf)

**Comment by fanquake** [comment]
ACK sdaftuar@65d384f

**Comment by laanwj** [comment]
ACK 65d384f, thanks for writing this

**Pull request merged**